4 view all available projects

nvenetucci opened the pull request:
Abstracts the code in the return block of `app/page.tsx` (Home page) into separate components: `ProjectGrid` and `ProjectCard`. Together, these components work to display a grid of project cards given that `ProjectGrid` is supplied a list of projects (`Project[]`) as a prop.

The app retains the same functionality as before, mainly just a refactor and visual differences. The styles were modified using Tailwind CSS in the TSX code. I tried to ensure the resulting view is responsive, but encourage reviewers to test window resizing on their end to confirm (e.g., try posting a project with a very long description).

Also, the 'Apply' button is just a placeholder and doesn't do anything for now. Let me know what you think of the appearance and if you'd like anything to look different!

nvenetucci committed:
Add components folder and ProjectGrid component

nvenetucci committed:
Add ProjectCard component

nvenetucci committed:
Add an object prop of type Project to the ProjectCard component

nvenetucci committed:
Fetch projects in ProjectGrid and map each to a ProjectCard

nvenetucci committed:
Change ProjectGrid to accept a prop of type Project[]

nvenetucci committed:
Add Link for ProjectCard title and minor style changes

nvenetucci committed:
Substitute ProjectGrid in return block of app/page.tsx

nvenetucci committed:
Remove unused imports and user variable from app/page.tsx

nvenetucci committed:
Remove async keyword from ProjectGrid function signature

nvenetucci committed:
Modify ProjectCard CSS to fix spacing when using longer description

daniiltsioma reviewed (changes requested):
Good use of abstraction! 

Couple quick things: in the class on Thursday you showed the example where the website was in dark mode, right now I see dark cards on light background:

<img width="1444" alt="image" src="https://github.com/user-attachments/assets/9b35a15a-d207-4d42-a9ce-3b59fc084fb1" />

I think we just need to choose one thing, either it is dark or light. 
Also, it would look nice for the grid to be aligned with the title on the left.

Responsive design looks perfect! I appreciate you spending time making it responsive.

nvenetucci committed:
Modify CSS colors to use dark theme

daniiltsioma commented:
Dark mode looks good! What do you think about aligning left side of the grid with Hobby Dev Hub title?

nvenetucci committed:
Modify form input outline color on focus and other CSS color changes

nvenetucci committed:
Remove container div from ProjectCard and add padding 8 to grid div

nvenetucci commented:
Alright, I think I fixed it. Gave the grid a padding of 8 to match the title. I also tried padding 7 and think that could probably work as well. Let me know what you think!

daniiltsioma reviewed (approved):
Looks good to me!

daniiltsioma commented:
I see there are some conflicts, I'll let you and @lbrescia figure out how to connect your code – Logan worked on the database part, now we need to connect his code with your front end

nvenetucci committed:
Merge branch 'main' into 4-view-all-available-projects

nvenetucci committed:
Add back in dummy database for Projects

nvenetucci commented:
Hey I fixed the merge conflicts since pull request #13 was merged into main this morning. Part of #13 included modifying `api/projects/` and removing the dummy database. My feature in this pull request was using that dummy db, so I added it back in as `api/dummy-db/`, see commit [7dbd79c](https://github.com/daniiltsioma/hobby-dev-hub/pull/15/commits/7dbd79c0d4541f89fbc83f87d58c33c0d214d800).

The dummy-db is temporary for now just to get the frontend working. Will need to create a new issue to get the frontend working with the MongoDB setup that was added in #13 .

@daniiltsioma @lbrescia Do you guys wants to take another look at this PR before merging?

nvenetucci commented:
Chatted with @daniiltsioma on new changes. I checked the routes and they all seem to be working. `api/test-db` looks to be working as well. Going to go ahead and merge.
